clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusInstanceCachingAgent.java

```diff
@@ -0,0 +1,280 @@
+    Long jobNamesStartTime = System.currentTimeMillis();
+    Map<String, String> jobNames = titusClient.getAllJobNames();
+    PercentileTimer
+      .get(registry, metricId.withTag("operation", "getAllJobNames"))
+      .record(System.currentTimeMillis() - jobNamesStartTime, MILLISECONDS);
+
+
+    Map<String, CacheData> serverGroups = createCache();
+    Map<String, CacheData> instances = createCache();
+
+    for (Task task : tasks) {
+      InstanceData data = new InstanceData(task, jobNames.get(task.getJobId()), account.getName(), region.getName());
+      cacheInstance(data, instances);
+      cacheServerGroup(data, serverGroups);
+    }
+
+    log.info("Caching {} instances in {}", instances.size(), getAgentType());
+    log.info("Caching {} server groups in {}", serverGroups.size(), getAgentType());
+
+    Map<String, Collection<CacheData>> cacheResult = new HashMap<>();
+    cacheResult.put(INSTANCES.ns, instances.values());
+    cacheResult.put(SERVER_GROUPS.ns, serverGroups.values());
+
+    PercentileTimer
+      .get(registry, metricId.withTag("operation", "loadData"))
+      .record(System.currentTimeMillis() - startTime, MILLISECONDS);
+    log.info("Caching completed in {}s in {}", MILLISECONDS.toSeconds(System.currentTimeMillis() - startTime), getAgentType());
+
+    return new DefaultCacheResult(cacheResult);
+  }
+
+  private void cacheInstance(InstanceData data, Map<String, CacheData> instances) {
+    CacheData instanceCache = instances.getOrDefault(data.instanceId, new MutableCacheData(data.instanceId));
+    instanceCache.getAttributes().putAll(objectMapper.convertValue(data.task, ATTRIBUTES));
+    instanceCache.getAttributes().put(HEALTH.ns, Collections.singletonList(getTitusHealth(data.task)));
+    instanceCache.getAttributes().put("task", data.task);
+    instanceCache.getAttributes().put("jobId", data.jobId);
+
+    if (!data.serverGroup.isEmpty()) {
+      instanceCache.getRelationships().computeIfAbsent(SERVER_GROUPS.ns, key -> new HashSet<>()).add(data.serverGroup);
+    } else {
+      instanceCache.getRelationships().computeIfAbsent(SERVER_GROUPS.ns, key -> new HashSet<>()).clear();
+    }
+    instances.put(data.instanceId, instanceCache);
+  }
+
+  private void cacheServerGroup(InstanceData data, Map<String, CacheData> serverGroups) {
+    if (!data.serverGroup.isEmpty()) {
+      CacheData serverGroupCache = serverGroups.getOrDefault(data.serverGroup, new MutableCacheData(data.serverGroup));
+      serverGroupCache.getRelationships().computeIfAbsent(INSTANCES.ns, key -> new HashSet<>()).add(data.instanceId);
+      serverGroups.put(data.serverGroup, serverGroupCache);
+    }
+  }
+
+  private Map<String, String> getTitusHealth(Task task) {
+    TaskState taskState = task.getState();
+    HealthState healthState = HealthState.Unknown;
+    if (Arrays.asList(TaskState.STOPPED, TaskState.FAILED, TaskState.CRASHED, TaskState.FINISHED, TaskState.DEAD, TaskState.TERMINATING).contains(taskState)) {
```

Comment:
I'd move this into a `static final` on the class, rather than creating the array every time we invoke the method. Same goes for the one below.